apps/tangle-dapp/src/containers/TopNavigationPanel.tsx

```diff
@@ -1,16 +1,21 @@
 import ConnectWalletButton from '@tangle-network/tangle-shared-ui/components/ConnectWalletButton';
 import { type FC } from 'react';
-
-import NetworkSelectionButton from '../components/NetworkSelectorButton';
-import TxHistoryDrawer from '../components/TxHistoryDrawer';
+import { useLocation } from 'react-router';
+import { PagePath } from '../types';
+import BridgeConnectWallet from '../components/bridge/BridgeConnectWallet';
+import BridgeNetworkDisplay from '../components/bridge/BridgeNetworkDisplay';
+import NetworkSelectorDropdown from '@tangle-network/tangle-shared-ui/components/NetworkSelectorDropdown';
+// import TxHistoryDrawer from '../components/TxHistoryDrawer';
 
 const TopNavigationPanel: FC = () => {
-  return (
-    <div className="flex items-center gap-2">
-      <NetworkSelectionButton />
+  const location = useLocation();
 
-      <ConnectWalletButton />
+  const isInBridgePath = location.pathname.startsWith(PagePath.BRIDGE);
 
+  return (
+    <div className="flex items-center gap-2">
+      {isInBridgePath ? <BridgeNetworkDisplay /> : <NetworkSelectorDropdown />}
```

Comment:
The `BridgeNetworkDisplay` and `NetworkSelectorDropdown` components exhibit significant code duplication in functionality, UI elements, and logic. We should unify them into a single component. You can provide the code for both components and ask AI to consolidate them.

Reply:
@AtelyPham `NetworkSelectorDropdown` is a component in `tangle-shared-ui`. To unify this we would have to pass a bridge specific props to `NetworkSelectorDropdown`.

Reply:
Is it `selectedSourceChain`? Can we use it to determine the logic and pass that prop to the `NetworkSelectorDropdown`? This way, props can be passed to components for behavior customization, keeping them generalized rather than feature-specific. You can also ask AI to assist with this.

Reply:
Yes, we need `selectedSourceChain` in `NetworkSelectorDropdown`. This is how it was previously implemented. We had a prop called `isBridgePath` on `NetworkSelectorDropdown`. We discussed not passing page-specific props (such as bridge, nomination, etc.) to components in `tangle-shared-ui`.

Reply:
We aren't passing page-specific props, but we can pass `isWrongEvmNetwork` and `switchToCorrectEvmChain`, which are not page-specific.